src/conventions/h-group/take-action.js

```diff
@@ -271,9 +260,14 @@ export function take_action(game) {
 	}
 
 	// Get a high value play clue involving next player (otherwise, next player can give it)
-	if (best_play_clue?.result.finesses.length > 0 && best_play_clue.result.finesses.some(f => f.playerIndex === nextPlayerIndex))
+	// TODO: We should probably require a higher value to be worth cluing while finessed.
```

Comment:
Open to thoughts on what sufficient value might be. It might make sense to make some of these decisions based on the pacing / clue efficiency required. At least in the cases where the clue wasn't already filtered as losing the finesse connection it's because it couldn't have been given by another player in time.

Reply:
I think it's more important to make sure we aren't sacrificing too much tempo. For example, I think it's bad to defer a finesse if:
- you're giving a *Selfish Finesse* (a finesse that uses cards in your own hand)
- the deferred finesse needs multiple rounds to resolve (we need to play multiple cards, or the connecting cards are out of order)
- endgame tempo requires us to play

Reply:
Okay, I think I've roughly added the conditions you've suggested but please take a look and let me know if i got any of them wrong.